processing/forward_handler.go

```diff
@@ -210,19 +210,13 @@ func (fh *ForwardHandler) Consume(e *types.Entry) error {
 				return err
 			}
 		}
-		for k, v := range fh.AddedFields {
-			val, err := util.EscapeJSON(v)
-			if err != nil {
-				fh.Logger.Warningf("cannot escape value: %s", v)
-				continue
-			}
-			newJSON, err := jsonparser.Set([]byte(e.JSONLine), val, k)
-			if err != nil {
-				fh.Logger.Warningf("cannot set %s: %s", k, v)
-				continue
-			} else {
-				e.JSONLine = string(newJSON)
-			}
+		// if length is 1 then there are no added fields, only a '}'
```

Comment:
Please comment/document in the code better what happens here.

From looking at it i guessed that you are getting rid of the comma (json-seperator), to 
add a field j += fmt.Sprintf(",%s:%s", kval, vval) afterwards.

Question that immediately arise from my side: Why isn't there a comma at the end added again?
How do we ensure that the json always has the form. This immediately damages the JSON when key values are not seperated by newline or when the last rune is a comma).

Reply:
Is this change tested appopriately?

Reply:
> From looking at it i guessed that you are getting rid of the comma (json-seperator), to
add a field j += fmt.Sprintf(",%s:%s", kval, vval) afterwards.

No, I am removing the final `}` brace to extend the JSON with `,"foo":"bar","baz":"quux"}`. If there are no added fields, the string that would be added is just the `}` again (hence only length 1),

Reply:
> Is this change tested appopriately?

This is covered by the existing test for the `add-fields` functionality, which still passes (which was my goal ;))

Reply:
:top:

Reply:
Comments added.